mod/src/main/java/holiday/server/mixin/ServerListMixin.java

```diff
@@ -0,0 +1,27 @@
+package holiday.server.mixin;
+
+import holiday.server.ClientEntrypoint;
+import net.minecraft.client.network.ServerInfo;
+import net.minecraft.client.option.ServerList;
+import org.spongepowered.asm.mixin.Final;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.Shadow;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;
+
+import java.util.List;
+
+@Mixin(ServerList.class)
+public class ServerListMixin {
+	@Shadow
+	@Final
+	private List<ServerInfo> servers;
+
+	@Inject(method = "loadFile", at = @At("TAIL"))
```

Comment:
Guessing the cause here, loadFile must not be being called if a `servers.dat` file doesn't exist. Might need a different approach, or a way to create the file.